### PR TITLE
Add network interface filter example in samples

### LINKED ISSUE #1170: [FEATURE] Documentation on How to Whitelist Interfaces
It would be nice to provide documentation and a code snippet on how to whitelist device interfaces.

### PR DESCRIPTION
**Pull request by disa6302**
*Issue #, if available:*
#1170 

*Description of changes:*
- Adding network interface filtering sample callback. 
- Updated README with instructions as well.

Resolves #1170
By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by unicornss — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -280,6 +280,12 @@ Similar to the heap profile, you only need to specify the following environment
 
 More information about what environment variables you can configure can be found [here](https://gperftools.github.io/gperftools/cpuprofile.html)
 
+### Filtering network interfaces
+
+This is useful to reduce candidate gathering time when it is known for certain network interfaces to not work well. A sample callback is available in `Common.c`. The `iceSetInterfaceFilterFunc` in `KvsRtcConfiguration` must be set to the required callback. In the sample, it can be done this way in `initializePeerConnection()`: 
+`configuration.kvsRtcConfiguration.iceSetInterfaceFilterFunc = sampleFilterNetworkInterfaces`
+
```
[the remark this answers is not in the document]
Comment (disa6302) [reply README.md:287]: I thought of doing that, but, if the code changes anytime, the lines would shift and the purpose will not be resolved. It is often easy to forget to update the README with the correct line numbers.

**Review by MushMal — APPROVED** [review]

**Review by hassanctech — APPROVED** [review]
ok

**Comment by disa6302** [comment]
Weirdly, Windows and static build fail. For static build, I suspect alpine updates to 10.3.1 that causes C compiler to break. Need to test with removing `alpine apk update` command to maintain stable 10.2.1 version.

**Comment by hassanctech** [comment]
> Weirdly, Windows and static build fail. For static build, I suspect alpine updates to 10.3.1 that causes C compiler to break. Need to test with removing `alpine apk update` command to maintain stable 10.2.1 version.

Looks like both with and without the `apline sdk update` it's using `10.3.1` of the compiler, so it is still failing.

File: .travis.yml [hunk 2]
```diff
@@ -153,10 +153,11 @@ matrix:
         - echo '{"ipv6":true,"fixed-cidr-v6":"2001:db8:1::/64"}' | sudo tee /etc/docker/daemon.json
         - sudo service docker restart
         - mkdir build
-        - docker run -e AWS_ACCESS_KEY_ID -e AWS_SECRET_ACCESS_KEY -w /src/build -dit --name alpine -v $(pwd):/src alpine
+        - docker run -e AWS_ACCESS_KEY_ID -e AWS_SECRET_ACCESS_KEY -w /src/build -dit --security-opt=seccomp=default.json --name alpine -v $(pwd):/src alpine:latest
```
Comment (hassanctech) [inline .travis.yml:156]: I think you need quotes here like this: :  --security-opt="seccomp=default.json"

**Pull request merged**